File: library/src/main/java/com/github/badoualy/ui/listener/FragmentTransactionHandler.java
package com.github.badoualy.ui.listener;


import android.app.Fragment;
import android.app.FragmentManager;

public interface FragmentTransactionHandler {

    void executeTransaction(FragmentTransactionRunnable transactionRunnable);

    void displayFragment(Fragment fragment);

    void displayFragment(Fragment fragment, boolean addToBackStack);

    void displayFragment(Fragment fragment, boolean addToBackStack, boolean clearStack);

    /** A custom runnable that will operate a FragmentTransaction */
    interface FragmentTransactionRunnable {
        void run(FragmentManager fm, int fragmentId);
    }
}

File: library/src/main/java/com/github/badoualy/ui/listener/FragmentTransactionHandler.kt
package com.github.badoualy.ui.listener


import android.app.Fragment
import android.app.FragmentManager

interface FragmentTransactionHandler {

    fun executeTransaction(transactionRunnable: FragmentTransactionRunnable)

    fun displayFragment(fragment: Fragment, addToBackStack: Boolean = true, clearStack: Boolean = false)

    /** A custom runnable that will operate a FragmentTransaction  */
    interface FragmentTransactionRunnable {
        fun run(fm: FragmentManager, fragmentId: Int)
    }
}
